spec/spec_helper.rb

```diff
@@ -223,6 +223,14 @@
     end
   end
 
+  # Permit allowing requests to specified external domains (which we block by
+  # default; see url_whitelist in Cuprite::DomainRestrictedDriver).
+  config.before do |example|
+    (example.metadata[:allowed_domains] || []).each do |allowed_domain|
+      page.driver.browser.url_allowlist << %r{\Ahttps://#{allowed_domain}/}
```

Comment:
I tested and found that the `page.driver.browser.url_allowlist` gets reset with each spec. In other words, an allowed domain for one example will not continue to be allowed when a subsequent example is executed. (This is good.)